autometa/taxonomy/ncbi.py

```diff
@@ -92,332 +93,305 @@ def __str__(self):
         return merged
 
-    def is_common_ancestor(self, parent_taxid, child_taxid):
-        """Determines whether the provided taxids have a non-root common ancestor
+    def is_common_ancestor(self, taxid_A, taxid_B):
+        """
+        Determines whether the provided taxids have a non-root common ancestor
 
         Parameters
         ----------
-        parent_taxid : int
-            Description of parameter `parent_taxid`.
-        child_taxid : int
-            Description of parameter `child_taxid`.
+        taxid_A : int
+            identifer for a taxon in NCI taxonomy databses - `nodes.dmp`, `names.dmp` or `merged.dmp`
+        taxid_B : int
+            identifer for a taxon in NCI taxonomy databses - `nodes.dmp`, `names.dmp` or `merged.dmp`
 
         Returns
         -------
         boolean
-            True if taxids share a common ancestor and False otherwise
+            True if taxids share a common ancestor else False
         """
-        ancestor_taxid = child_taxid
-        while ancestor_taxid != 1:
-            if parent_taxid == ancestor_taxid:
-                return True
-            ancestor_taxid = self.parent(ancestor_taxid)
-        return False
-
-
-def main():
-    import argparse
-    import logging as logger
-    logger.basicConfig(
-        format='%(asctime)s : %(name)s : %(levelname)s : %(message)s',
-        datefmt='%m/%d/%Y %I:%M:%S %p',
-        level=logger.DEBUG)
-    parser = argparse.ArgumentParser(
-        description='Autometa NCBI utilities class')
-    parser.add_argument('ncbi', help='</path/to/ncbi/database/directory>')
-    parser.add_argument(
-        '--query-taxid',
-        help='query taxid to test NCBI class functionality',
-        default=1222,
-        type=int,
-    )
-    parser.add_argument('--verbose', help="add verbosity",
-                        action='store_true', default=True)
-    args = parser.parse_args()
-    ncbi = NCBI(args.ncbi, args.verbose)
-    query_taxid_parent = ncbi.parent(taxid=args.query_taxid)
-    logger.info(
-        f'{args.query_taxid} Name: {ncbi.name(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Order: {ncbi.name(taxid=args.query_taxid, rank="order")}\n'
-        f'{args.query_taxid} Class: {ncbi.name(taxid=args.query_taxid, rank="class")}\n'
-        f'{args.query_taxid} Rank: {ncbi.rank(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} Lineage:\n{ncbi.lineage(taxid=args.query_taxid)}\n'
-        f'{args.query_taxid} is_common_ancestor {query_taxid_parent}: '
-        f'{ncbi.is_common_ancestor(parent_taxid=query_taxid_parent, child_taxid=args.query_taxid)}\n'
-    )
-
-
-if __name__ == '__main__':
-    main()
+        A_lineage_taxids = {taxids.get("taxid")
+                            for taxids in self.lineage(taxid_A)}
+        B_lineage_taxids = {taxids.get("taxid")
+                            for taxids in self.lineage(taxid_B)}
+        common_ancestor = B_lineage_taxids.intersection(A_lineage_taxids)
```

Comment:
I've already made this comment, but it appears it either has been removed or does not show in a new review. But it was not addressed.

When calling `self.lineage`, you are retrieving one ancestor at a time, not `taxids`. 

Perhaps:

```suggestion
        lineage_a_taxids = {ancestor.get("taxid")
                            for ancestor in self.lineage(taxid_A)}
        lineage_b_taxids = {ancestor.get("taxid")
                            for ancestor in self.lineage(taxid_B)}
        common_ancestor = lineage_b_taxids.intersection(lineage_a_taxids)
```